socket/socket.go

```diff
@@ -71,6 +63,31 @@ func (w *Socket) Read() {
 	}
 }
 
+// SendSuccess sends a success message
+func (w *Socket) SendSuccess(msg interface{}) {
+	w.Send(types.BaseMessage{
+		Status: types.StatusSuccess,
+		Data:   msg,
+	})
+}
+
+// SendErrorMessage sends the error message
+func (w *Socket) SendErrorMessage(err error) {
+	w.Send(types.BaseMessage{
+		Status: types.StatusFailure,
+		Error:  err.Error(),
+	})
+}
+
+// SendDataOrError sends the appropriate data or error
+func (w *Socket) SendDataOrError(data interface{}, err error) {
```

Comment:
This is never used, if we are going to include it I would change below to call it, but I don't think it is necessary. I would stick with just the two above imo

Reply:
Ah I see. The `BaseAPI.Send()` function provides the same functionality. I can get rid of either `BaseAPI.Send()` or `Socket.SendDataOrError()`. Which do you think would be cleaner to keep?

Reply:
Ah nvm, just read your comment below